src/erlcloud_ddb2.erl

```diff
@@ -3767,6 +4279,141 @@ update_table(Table, ReadUnits, WriteUnits, Opts, Config) ->
+    {<<"IndexName">>, IndexName};
+dynamize_global_secondary_index_auto_scaling_update_opt({provisioned_write_capacity_auto_scaling_update, ProvisionedWriteCapacityAutoScalingUpdate}) ->
+    {<<"ProvisionedWriteCapacityAutoScalingUpdate">>, dynamize_maybe_list(fun dynamize_auto_scaling_settings_update_opt/1,
+                                                                          ProvisionedWriteCapacityAutoScalingUpdate)}.
+
+-spec dynamize_global_secondary_index_auto_scaling_update_opts([replica_global_secondary_index_auto_scaling_opts()]) -> jsx:json_term().
+dynamize_global_secondary_index_auto_scaling_update_opts(GlobalSecondaryIndexUpdates) ->
+    [dynamize_maybe_list(fun dynamize_global_secondary_index_auto_scaling_update_opt/1,
+                         Update) || Update <- GlobalSecondaryIndexUpdates].
+
+-spec dynamize_replica_global_secondary_index_auto_scaling_update_opt(replica_global_secondary_index_auto_scaling_opt()) -> json_pair().
+dynamize_replica_global_secondary_index_auto_scaling_update_opt({index_name, IndexName}) ->
+    {<<"IndexName">>, IndexName};
+dynamize_replica_global_secondary_index_auto_scaling_update_opt({provisioned_read_capacity_auto_scaling_update, ProvisionedReadCapacityAutoScalingUpdate}) ->
+    {<<"ProvisionedReadCapacityAutoScalingUpdate">>, dynamize_auto_scaling_settings_update_opts(ProvisionedReadCapacityAutoScalingUpdate)}.
+
+-spec dynamize_replica_global_secondary_index_auto_scaling_update_opts(replica_global_secondary_index_auto_scaling_opts()) -> jsx:json_term().
+dynamize_replica_global_secondary_index_auto_scaling_update_opts(Opts) ->
+    dynamize_maybe_list(fun dynamize_replica_global_secondary_index_auto_scaling_update_opt/1,
+                        Opts).
+
+-spec dynamize_replica_auto_scaling_update(replica_auto_scaling_update_opt()) -> json_pair().
+dynamize_replica_auto_scaling_update({region_name, RegionName}) ->
+    {<<"RegionName">>, RegionName};
+dynamize_replica_auto_scaling_update({replica_global_secondary_index_updates, ReplicaGlobalSecondaryIndexUpdates}) ->
+    {<<"ReplicaGlobalSecondaryIndexUpdates">>, [dynamize_replica_global_secondary_index_auto_scaling_update_opts(Update) || Update <- ReplicaGlobalSecondaryIndexUpdates]};
+dynamize_replica_auto_scaling_update({replica_provisioned_read_capacity_auto_scaling_update, ReplicaProvisionedReadCapacityAutoScalingUpdate}) ->
+    {<<"ReplicaProvisionedReadCapacityAutoScalingUpdate">>, dynamize_auto_scaling_settings_update_opts(ReplicaProvisionedReadCapacityAutoScalingUpdate)}.
+
+-spec dynamize_replica_auto_scaling_updates([replica_auto_scaling_update_opts()]) -> jsx:json_term().
+dynamize_replica_auto_scaling_updates(ReplicaUpdates) ->
+    [dynamize_maybe_list(fun dynamize_replica_auto_scaling_update/1,
+                         Update) || Update <- ReplicaUpdates].
+
+-spec maybe_update_config_timeout(aws_config(), MinDesiredTimeout :: pos_integer()) -> aws_config().
+maybe_update_config_timeout(Config, MinDesiredTimeout) ->
```

Comment:
@nalundgaard I think this matches what we discussed. I just realized I left it down here instead of moving to the "shared" section towards the top, I can move it if you want it to remain a dedicated function.

Reply:
I don't guess it matters much. @motobob curious if you have an opinion. this is basically needed to avoid the default timeout from preventing requests to this endpoint from ever working.